src/experimental/patterns/ec2-app.ts

```diff
@@ -355,9 +372,36 @@ export class GuUserDataForRollingUpdateExperimental {
 
       echo "Instance running build ${buildIdentifier} is healthy in target group."
       `,
-      `# ${GuEc2AppExperimental.name} UserData End`,
+      `# ${GuEc2AppExperimental.name} Instance Health Check End`,
     );
 
+    if (slowStartDuration) {
+      const slowStartInSeconds = slowStartDuration.toSeconds().toString();
+      targetGroup.setAttribute("slow_start.duration_seconds", slowStartInSeconds);
+      autoScalingGroup.userData.addCommands(
+        `# ${GuEc2AppExperimental.name} SlowStart Wait Period Start`,
+        `
+        echo "Sleeping for ${slowStartInSeconds} seconds while instance running build ${buildIdentifier} warms up..."
+        sleep ${slowStartInSeconds}s
+        echo "Instance running build ${buildIdentifier} should have warmed up by now..."
+
+        STATE=$(aws elbv2 describe-target-health \
+        --target-group-arn ${targetGroup.targetGroupArn} \
+        --region ${region} \
+        --targets Id=$INSTANCE_ID,Port=${applicationPort} \
+        --query "TargetHealthDescriptions[0].TargetHealth.State")
+
+        if [ "$STATE" == "\\"healthy\\"" ]; then
```

Comment:
Note that we do a one-off check here rather than polling.

If an instance is not healthy after it's warm-up period[^1] then this means that it has:

1) Initially launched correctly and become healthy
2) Fallen over despite only being asked to deal with a lower-than-average share of traffic

Consequently we just assume that something is wrong with the new build at this point and fail the deployment.

[^1]: In this scenario the instance will have [exited slow-start mode](https://docs.aws.amazon.com/elasticloadbalancing/latest/application/edit-target-group-attributes.html#slow-start-mode)[^1]; in theory it could become healthy again and re-enter slow start mode, but we are not waiting to find out if this happens.